scripts/Interface/ui/sans_isis/sans_data_processor_gui.py

```diff
@@ -0,0 +1,1542 @@
+from __future__ import (absolute_import)
+import os
+try:
+    from mantidplot import *
+    canMantidPlot = True
+except ImportError:
+    canMantidPlot = False
+
+from . import ui_sans_data_processor_window as ui_sans_data_processor_window
+from PyQt4 import QtGui, QtCore
+from mantid.simpleapi import *
+from mantid.kernel import (Logger, config)
+from abc import ABCMeta, abstractmethod
+from six import with_metaclass
+from inspect import isclass
+from mantidqtpython import MantidQt
+from sans.common.enums import (ReductionDimensionality, OutputMode, SaveType, SANSInstrument,
+                               RangeStepType, SampleShape, ReductionMode, FitType)
+from sans.gui_logic.gui_common import (get_reduction_mode_from_gui_selection,
+                                       get_string_for_gui_from_reduction_mode)
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Globals
+# ----------------------------------------------------------------------------------------------------------------------
+gui_logger = Logger("SANS GUI LOGGER")
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Free Functions
+# ----------------------------------------------------------------------------------------------------------------------
+def open_file_dialog(line_edit, filter_text, directory):
+    dlg = QtGui.QFileDialog()
+    dlg.setFileMode(QtGui.QFileDialog.AnyFile)
+    dlg.setFilter(filter_text)
+    dlg.setDirectory(directory)
+    if dlg.exec_():
+        file_names = dlg.selectedFiles()
+        if file_names:
+            line_edit.setText(file_names[0])
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Gui Classes
+# ----------------------------------------------------------------------------------------------------------------------
+class SANSDataProcessorGui(QtGui.QMainWindow, ui_sans_data_processor_window.Ui_SansDataProcessorWindow):
+    data_processor_table = None
+    main_presenter = None
```

Comment:
I don't think this is ever set is it? The `__init__` method creates a new private `_main_presenter` variable, was this intended to be that?